### PR TITLE
Complete error handling for Share resources

### PR DESCRIPTION
**Pull request by davivel**

### CODE REVIEW

File: src/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.java [hunk 1]
```diff
@@ -141,45 +132,22 @@ protected RemoteOperationResult run(OwnCloudClient client) {
...
+						new ShareXMLParser()
+				);
+				parser.setOneOrMoreSharesRequired(true);
+				parser.setOwnCloudVersion(client.getOwnCloudVersion());
+				parser.setServerBaseUri(client.getBaseUri());
+				result = parser.parse(response);
+
+				if (result.isSuccess()) {
+					Log_OC.d(TAG, "Created " + result.getData().size() + " share(s)");	// should be one
+					if (mGetShareDetails) {
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.java]: The two if statements could be merged into one with && except when the log statement is important to be done anyways not depending on the second condition.

File: src/com/owncloud/android/lib/resources/shares/ShareToRemoteOperationResultParser.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShareToRemoteOperationResultParser {
```
Comment (masensio) [inline src/com/owncloud/android/lib/resources/shares/ShareToRemoteOperationResultParser.java]: This new kind of error helps to consider more error messages for sharing operations.
:+1:

**Comment by davivel** [comment]
Updated after review.

Please, notice also https://github.com/owncloud/android/pull/1216

**Comment by masensio** [comment]
Code reviewed.
Waiting for Travis before merging

**Pull request merged**